### DISCUSSION TITLE
Запуск в многопоточном режиме.

### DISCUSSION THREAD

**Discussion by TempoSky** [bol-van/zapret#633] [Q&A] [answered]
Добрый день. 
Хочу поинтересоваться, есть ли возможность запускать nfqws в несколько потоков для увеличения производительности ПО?

**Comment by bol-van** [comment, marked as answer]
нет, но можно разделить очереди средствами ip/nf tables и запустить несколько инстансов
там можно делать балансировку
в этом случае придется сделать через коннмарк так, чтобы входящие и исходящие от одного соединения шли на один инстанс

**Reply by TempoSky** [reply]
Получается чтобы матчить соединения нужно будет несколько  wan и lan интерфейсов поднимать?

**Reply by bol-van** [reply]
Зачем ? Я не настраивал это ни разу, но теоретически это делается так примерно
Берется некоторое значение. Например, source port. Он достаточно случаен.
Выбираются из него младшие биты. Скажем, 2 младших бита дает 4 варианта.
Ставится коннмарк по ним.
Пишутся правила редиректа с учетом коннмарк как на выход, так и на вход. Каждая комбинация на свою очередь.
В nftables можно в одно правило запихать, используя сеты
Так уже кто-то делал. Работало
